### ISSUE TITLE
Object-block ambiguity

### ISSUE THREAD

**Issue by wcjohnson** [wcjohnson/lightscript#6]
```js
f() -> {}
if x: {}
```
```js
function f() {}

if (x) {
}
```
These should be making empty objects, not empty blocks.

**Comment by wcjohnson** [comment]
In the case of `if`, we can disambiguate in the following way:

Object: `if cond: {...}`
Block: `if cond {...}` (note the absence of `:`)

This doesn't quite work for arrow functions, though, because there is no disambiguating `:` token.

So, idea: `arrowFunctionWhiteblockOnly` flag (implies LSC) which would outlaw braceblock arrow functions. Then a braceblock at opening position in an arrow function can always be treated as an object.

The LSC escape valve of `;{` to forcibly open a braceblock should be left intact.

**Comment by wcjohnson** [comment]
- If a whiteblock begins with `{`, try to parse a single object
-Tests:
```js
f() => {x}; y
```
```js
f() =>
  {x}
```
```js
f() =>
  {x}
  y
```
```js
f() =>
  {x};
  y
```

**Comment by wcjohnson** [comment]
Look at:
- es2015/uncategorized/177/expected.json: comment node moved

**Comment by wcjohnson** [comment]
Implemented in 2.1.0 via `preferObjectLiteral` flag
